backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt

```diff
@@ -0,0 +1,266 @@
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.LayoutDirection
+import net.skyscanner.backpack.compose.flightleg.internal.BpkAccessoryTitleDetails
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.Aircraft
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun BpkFlightLeg(
+  departureArrivalTime: String,
+  description: AnnotatedString,
+  stopsInfo: AnnotatedString,
+  duration: String,
+  modifier: Modifier = Modifier,
+  nextDayArrival: String? = null,
+  operatedBy: String? = null,
+  warning: String? = null,
+  contentDescription: String? = null,
+  carrierLogo: @Composable (BoxScope.() -> Unit)? = null,
+) {
+  Box(modifier = modifier
+    .fillMaxWidth()
+    .semantics(mergeDescendants = true) {
+      if (contentDescription == null) {
+        // prevent screen readers from reading this allowing a broader
+        // content description to be set on the parent node
+        invisibleToUser()
+      } else {
+        this.contentDescription = contentDescription
+      }
+    },
+  ) {
+    BpkAccessoryTitleDetails(
+      modifier = Modifier.fillMaxWidth(fraction = 0.67f),
+      title = buildAnnotatedString {
+        append(departureArrivalTime)
+        nextDayArrival?.let {
+          withStyle(
+            SpanStyle(
+              color = BpkTheme.colors.textPrimary,
+              fontSize = BpkTheme.typography.label3.fontSize,
+              baselineShift = BaselineShift.Superscript,
+            ),
+          ) {
+            append(it)
+          }
+        }
+      },
+      details = mutableListOf<AnnotatedString>().apply {
+        add(description)
+        operatedBy?.let {
+          add(AnnotatedString(it))
+        }
+        warning?.let {
+          add(
+            buildAnnotatedString {
+              withStyle(
+                SpanStyle(
+                  color = BpkTheme.colors.textError,
+                  fontSize = BpkTheme.typography.caption.fontSize,
+                ),
+              ) {
+                append(it)
+              }
+            },
+          )
+        }
+      },
+      accessoryView = carrierLogo,
+    )
+    BpkAccessoryTitleDetails(
+      modifier = Modifier.align(Alignment.TopEnd),
+      title = stopsInfo,
+      textAlignment = Alignment.End,
+      titleStyle = BpkTheme.typography.label3,
+      details = listOf(
+        AnnotatedString(duration),
+      ),
+    )
+  }
+}
+
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO, showBackground = true, backgroundColor = 0xffffff)
+@Composable
+private fun LightModeSimpleDirect() {
```

Comment:
we generally don't provide previews within the component. also there's a new `BpkPreviews` annotation for use to provide the default, dm + rtl combination - let me share that in the compose channel :) )